### PR TITLE
Get dataset configuration

### PR DESCRIPTION
**Pull request by exarkun**
https://clusterhq.atlassian.net/browse/FLOC-1387

### CODE REVIEW

File: flocker/control/httpapi.py [hunk 1]
```diff
@@ -74,6 +74,29 @@ def version(self):
         """
         return {u"flocker":  __version__}
 
+    @app.route("/datasets", methods=['GET'])
```
Comment (itamarst) [inline flocker/control/httpapi.py]: I would suggest this and the `POST` get changed to `/configuration/datasets` as you suggested at some point, so they are more symmetric with `/state/datasets` and it's immediately clear one is referring to *configuration*, not state.

(Might want to change this again when we have API that can actually tell you when a change has been applied successfully, but that may be a while...)

**Comment by itamarst** [comment]
Additional things to fix:
1. Both this and the current state endpoint don't include the JSON schema in the generated documentation. I believe this is because our documentation generator doesn't handle arrays, and assumes output is always a JSON object/struct. File an issue separately for this, I guess.
2. Missing an acceptance test. This should be easy to write once FLOC-1397 is done. Up to you whether to do it in this branch or another, but we should require acceptance tests for each new API endpoint from here on out.

File: flocker/control/test/test_httpapi.py [hunk 2]
```diff
@@ -451,10 +451,146 @@ def get_dataset_ids(deployment):
...
+        expected = [
+            api_dataset_from_dataset_and_node(
+                manifestation_a.dataset, self.NODE_A
+            ),
+            api_dataset_from_dataset_and_node(
+                manifestation_b.dataset, self.NODE_A
+            ),
+        ]
+        return self._dataset_test(deployment, expected)
+
```
Comment (itamarst) [inline flocker/control/test/test_httpapi.py]: Maybe add unit test for dataset that has metadata, so we can see that the schema verification doesn't reject it.

**Comment by itamarst** [comment]
Please accept if you're doing acceptance test as separate issue, or reject if you're doing acceptance test in this branch.

**Comment by exarkun** [comment]
> 2. Missing an acceptance test. This should be easy to write once FLOC-1397 is done. Up to you whether to do it in this branch or another, but we should require acceptance tests for each new API endpoint from here on out.

I think this will probably make sense once we have a more sophisticated configuration storage system (ZooKeeper or what have you).  For now I think the acceptance test would basically exercise exactly the same code path as the integration test so I'm going to skip it.

**Pull request merged**